src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -315,6 +414,13 @@ private OAuthProxy(
 
                 this.expiryFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
             }
+
+            this.grant = grant;
+            if(this.grant != null)
```

Comment:
Why would `grant` be `null`, must it not always be `non-null`?
This would be preferable to limit the number of special cases.
If `subject` is `null` in `BEGIN`, perhaps we always need to create a new `OAuthAccessGrant` ?